offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/components/product/create/CreateProductView.groovy

```diff
@@ -1,14 +1,296 @@
+            ProductUnit newValue = it.newValue as ProductUnit
+            if (newValue) {
+                productUnitComboBox.value = newValue
+            } else {
+                productUnitComboBox.value = productUnitComboBox.emptyValue
+            }
+        })
+        productUnitComboBox.addSelectionListener({
+            createProductViewModel.setProductUnit(it.value as ProductUnit)
+        })
+
+        createProductViewModel.addPropertyChangeListener("productCategories", {
+            ProductCategory newValue = it.newValue as ProductCategory
+            if (newValue) {
+                productCategoriesComboBox.value = newValue
+            } else {
+                productCategoriesComboBox.value = productCategoriesComboBox.emptyValue
+            }
+        })
+        productCategoriesComboBox.addSelectionListener({
+            createProductViewModel.setProductCategories(it.value as ProductCategory)
+        })
+
+        /*
+       We listen to the valid properties. whenever the presenter resets values in the viewmodel
+       and resets the valid properties the component error on the respective component is removed
+       */
+        createProductViewModel.addPropertyChangeListener({
+            switch (it.propertyName) {
+                case "productNameValid":
+                    if (it.newValue || it.newValue == null) {
+                        productNameField.componentError = null
+                    }
+                    break
+                case "productDescriptionValid":
+                    if (it.newValue || it.newValue == null) {
+                        productDescriptionField.componentError = null
+                    }
+                    break
+                case "productUnitPriceValid":
+                    if (it.newValue || it.newValue == null) {
+                        productUnitPriceField.componentError = null
+                    }
+                    break
+                case "productUnitValid":
+                    if (it.newValue || it.newValue == null) {
+                        productUnitComboBox.componentError = null
+                    }
+                    break
+                case "productCategoryValid":
+                    if (it.newValue || it.newValue == null) {
+                        productCategoriesComboBox.componentError = null
+                    }
+                    break
+                default:
+                    break
+            }
+            createProductButton.enabled = allValuesValid()
+        })
+    }
+
+    /**
+     * This method adds validation to the fields of this view
+     */
+    private void setupFieldValidators() {
+
+        Validator<String> nameValidator =  Validator.from({String value -> (value && !value.trim().empty)}, "Please provide a valid name.")
+        Validator<String> numberValidator = new RegexpValidator("This is not a number!", "[-]?[0-9]*\\.?,?[0-9]+")
```

Comment:
While this is a very good implementation(cause i wrote it ;) ) We have to be careful since it allows "." as well as "," which may make problems when converting the input number into a double later

Reply:
```suggestion
        Validator<String> numberValidator = new RegexpValidator("This is not a number!", "[-]?[0-9]*\\.?[0-9]+")
```

Reply:
Commas have feelings too you know :/